app/models/notification.rb

```diff
@@ -10,4 +10,15 @@ def self.new_notification(sender_id, receiver_id)
     end
     notification.count
   end
+
+  def self.update_notifications_and_get_messages(current_user, sender_id)
+    notification = find_by(receiver_id: current_user.id, sender_id:)
+    notification.update(count: 0, read: true) if notification
+
+    messages = Message.where(sender_id: [current_user.id, sender_id],
+                             receiver_id: [current_user.id,
+                                           sender_id])
+
+    [notification, messages]
```

Comment:
- Do we need to return notifications here? 
- Messages should not be filtered here.